### PR TITLE
Fix: Nacl certificates default paths for windows

### PR DESCRIPTION
**Pull request by tacf**
### What does this PR do?
Add support for nacl on Windows masters.

### What issues does this PR fix or reference?
Nacl defaults didn't have source path for certificates based on the OS. 

### Tests written?
No
### Commits signed with GPG?
No

CC: @yagnik

### CODE REVIEW

**Review by twangboy — APPROVED** [review]

**Comment by rallytime** [comment]
@tacf There is a lint error on this change: https://jenkins.saltstack.com/job/PR/job/salt-pr-lint-n/18942/violations/file/salt/modules/nacl.py/

**Comment by tacf** [comment]
@rallytime Pylint is complaining about the backslack on paths, is it really a problem? I could add the raw tag for the string but imo is more of a fix for the linter than one for the sake of code readability and linting.

**Comment by rallytime** [comment]
@tacf Yeah, we will need that lint error fixed before we can merge this in.

**Comment by tacf** [comment]
@rallytime Is there anything else that i need to do?

File: salt/modules/nacl.py [hunk 1]
```diff
@@ -183,12 +184,13 @@ def _get_config(**kwargs):
     '''
     Return configuration
     '''
+    cert_path = r'C:\salt\conf\pki\master' if sys.platform == 'win32' else '/etc/salt/pki/master/'
```
Comment (lorengordon) [inline salt/modules/nacl.py]: Rather than hard coding these here, is the configured path value available through `__opts__['pki_dir']`?

Comment (tacf) [reply salt/modules/nacl.py]: how so?

Comment (lorengordon) [reply salt/modules/nacl.py]: Literally. 😄 

I'd probably just change the code to this form:

```
'sk_file': os.path.join(__opts__['pki_dir'], 'nacl'),
```

and remove the `cert_path` variable as it is unneeded.

Comment (tacf) [reply salt/modules/nacl.py]: ok, i'll look into it. Still getting familiar with the code.

**Comment by rallytime** [comment]
Hi @tacf - Did you get a chance to come back to this one?

**Comment by lorengordon** [comment]
That's how I'd do it. 😁 Minus the unnecessary `import sys`...

**Review by lorengordon — APPROVED** [review]

**Review by rallytime — APPROVED** [review]

**Comment by terminalmage** [comment]
This has already been partially fixed in an earlier release branch via https://github.com/saltstack/salt/pull/46426. I've opened https://github.com/saltstack/salt/pull/46639 to complete the fix.

But more importantly than that, this is an incorrect fix. `os.path.join(__opts__['pki_dir'], 'nacl')` will strip the `master` dir from the path and make the default location `/etc/salt/pki/nacl`.

Because this is an incorrect fix, and the fixes from earlier release branches will get merged forward and will therefore make their way into develop, I am going to close this.

**Comment by terminalmage** [comment]
I'll also note that @garethgreenaway's fix from #46426 had already been merged into develop anyway, which had caused this PR to have a merge conflict.

**Pull request closed without merging**